File: src/domain/lifecycle/target-layout.mjs
import path from "node:path";

export const metadataFile = ".architext.json";
export const legacyMetadataFile = ".architext-install.json";
export const instructionFiles = ["AGENTS.md", "CLAUDE.md"];
export const generatedIgnores = ["docs/architext/dist/", "docs/architext/.architext-write.lock/"];
export const copiedInstallEntries = [
  "AGENTS_APPENDIX.md",
  "LLM_ARCHITEXT.md",
  "README.md",
  "index.html",
  "dist",
  "node_modules",
  "package-lock.json",
  "package.json",
  "public",
  "schema",
  "src",
  "tools",
  "tsconfig.json",
  "vite.config.ts"
];
export const rootScripts = {
  architext: "architext serve .",
  "architext:build": "architext build .",
  "architext:clean": "architext clean .",
  "architext:doctor": "architext doctor .",
  "architext:prompt": "architext prompt .",
  "architext:validate": "architext validate ."
};

export function architextDir(target) {
  return path.join(target, "docs", "architext");
}

export function dataDir(target) {
  return path.join(architextDir(target), "data");
}

export function metadataPath(target) {
  return path.join(architextDir(target), metadataFile);
}

export function legacyMetadataPath(target) {
  return path.join(architextDir(target), legacyMetadataFile);
}

export function copiedInstallCandidatePaths(target) {
  return copiedInstallEntries.map((entry) => path.join(architextDir(target), entry));
}

File: crates/architext-cli/src/commands/sync/target_layout.rs
//! Mirrors `src/domain/lifecycle/target-layout.mjs` constants and path helpers.
//! Kept here (not in architext-core) because only the CLI sync command needs them.

use std::path::{Path, PathBuf};

pub const METADATA_FILE: &str = ".architext.json";
pub const LEGACY_METADATA_FILE: &str = ".architext-install.json";
pub const INSTRUCTION_FILES: &[&str] = &["AGENTS.md", "CLAUDE.md"];
pub const GENERATED_IGNORES: &[&str] = &[
    "docs/architext/dist/",
    "docs/architext/.architext-write.lock/",
];
pub const COPIED_INSTALL_ENTRIES: &[&str] = &[
    "AGENTS_APPENDIX.md",
    "LLM_ARCHITEXT.md",
    "README.md",
    "index.html",
    "dist",
    "node_modules",
    "package-lock.json",
    "package.json",
    "public",
    "schema",
    "src",
    "tools",
    "tsconfig.json",
    "vite.config.ts",
];

pub fn architext_dir(target: &Path) -> PathBuf {
    target.join("docs").join("architext")
}

pub fn data_dir(target: &Path) -> PathBuf {
    architext_dir(target).join("data")
}

pub fn metadata_path(target: &Path) -> PathBuf {
    architext_dir(target).join(METADATA_FILE)
}

pub fn legacy_metadata_path(target: &Path) -> PathBuf {
    architext_dir(target).join(LEGACY_METADATA_FILE)
}

pub fn copied_install_candidate_paths(target: &Path) -> Vec<PathBuf> {
    COPIED_INSTALL_ENTRIES
        .iter()
        .map(|entry| architext_dir(target).join(entry))
        .collect()
}
